### PR TITLE
Move azure pipelines embedded bash scripts to separate files

### PR DESCRIPTION
**Pull request by Winbobob**
Commit Message: Move azure pipelines embedded bash scripts to separate files
Additional Description: Fix bash warnings found by shellcheck.
- detect-repo-changes.sh
Screenshot 2024-07-07 at 00 24 27

- set-dev-release-envs.sh
Screenshot 2024-07-07 at 00 13 43

- decide-what-to-run.sh
Screenshot 2024-07-07 at 00 15 42

- decide-what-to-publish.sh
Screenshot 2024-07-07 at 00 21 56

Risk Level: Low
Testing:
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: N/A

### CODE REVIEW

**Review by phlax — COMMENTED** [review]
hi @Winbobob happy to land this - agree its a better pattern - but flagging this hasnt been done mostly as we are moving away from azp altogether

**Comment by Winbobob** [comment]
> hi @Winbobob happy to land this - agree its a better pattern - but flagging this hasnt been done mostly as we are moving away from azp altogether

Good to know! May I ask which CI service will we use next? And what is the ETA of the migration?

**Comment by phlax** [comment]
> May I ask which CI service will we use next?

github workflows

> And what is the ETA of the migration?

gradual/ongoing

File: .azure-pipelines/env/decide-what-to-publish.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    elif [[ $STATE_ISDEV == false ]]; then
+        if [[ $STATE_VERSIONPATCH -eq 0 ]]; then
+            # A just-forked branch
+            PUBLISH_GITHUB_RELEASE=false
+        fi
+    fi
+fi
+
+# Only run Envoy release CI if explictly set
+if { [[ "$PRESUBMIT" != 'true' ]] && [[ "$POSTSUBMIT" != 'true' ]]; } || [[ $BUILD_REASON == "Schedule" ]]; then
```
Comment (Winbobob) [inline .azure-pipelines/env/decide-what-to-publish.sh:21]: Not sure where are `PRESUBMIT`, `POSTSUBMIT`, and `NOSYNC` defined?

Comment (phlax) [reply .azure-pipelines/env/decide-what-to-publish.sh:21]: they are defined as azp library vars

Comment (Winbobob) [reply .azure-pipelines/env/decide-what-to-publish.sh:21]: Thanks, I assume `ISSTABLEBRANCH` and `ISMAIN` are also azure ado pipeline lib vars, right?

Comment (phlax) [reply .azure-pipelines/env/decide-what-to-publish.sh:21]: no - they are defined _in_ the pipeline - they are _not_ case sensitive - see, eg:

```
$ git grep -i ISSTABLEBRANCH
```

Comment (Winbobob) [reply .azure-pipelines/env/decide-what-to-publish.sh:21]: Ah, I see. The ado pipeline syntax is a bit subtle. Found the doc: https://learn.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch#access-variables-through-the-environment

Comment (Winbobob) [reply .azure-pipelines/env/decide-what-to-publish.sh:21]: I don't need to update the PR then, and the PR is ready to review, thanks!

**Comment by Winbobob** [comment]
Kindly bump, cc @phlax

**Review by phlax — APPROVED** [review]
lgtm, thanks @Winbobob

**Pull request merged**